Userland/Libraries/LibWeb/HTML/HTMLFormElement.cpp

```diff
@@ -1012,4 +1038,61 @@ WebIDL::ExceptionOr<JS::Value> HTMLFormElement::named_item_value(FlyString const
     return node;
 }
 
+// https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#default-button
+JS::GCPtr<HTML::HTMLButtonElement> HTMLFormElement::default_button()
+{
+    // A form element's default button is the first submit button in tree order whose form owner is that form element.
+    JS::GCPtr<HTML::HTMLButtonElement> default_button;
+
+    for_each_in_subtree_of_type<HTML::HTMLButtonElement>([&](auto& button) {
```

Comment:
I don't think this is quiteee right, because of the form attribute you can use to assign an element to a form owner and thus not have it in its subtree

Reply:
Ah good call. I found we store the elements associated with the form in a list already, so we can just iterate that.